tests/page-objects/default/sms/messages.wdio.page.js

```diff
@@ -1,57 +1,88 @@
+const getMessageHeader = async () => {
+  return {
+    name: await $(`${MESSAGE_HEADER} .name`).getText(),
+    phone: await $(`${MESSAGE_HEADER} .phone`).getText(),
+    lineage: await $(`${MESSAGE_HEADER} .horizontal.lineage`).getText()
+  };
 };
-const messageContentIndex = (index = 1) => $(`#message-content li:nth-child(${index})`);
-const messageContentText = (messageContentElement) => messageContentElement.$('.data p:first-child');
-const messageDetailStatus = async () => (await messageContentIndex()).$('.data .state.received');
-
-const messageText = text => $('#send-message textarea').setValue(text);
-const sendMessageModalSubmit = () => $('a.btn.submit:not(.ng-hide)');
-const messageRecipientSelect = () => $('#send-message input.select2-search__field');
-const contactNameSelector = () => $('.sender .name');
-const exportButton = () => $('.mat-mdc-menu-content .mat-mdc-menu-item[test-id="export-messages"]');
 
-const submitMessage = async () => {
+const getMessageContent = async  (index = 1) => {
+  const sms = await $(`${MESSAGE_CONTENT} li:nth-child(${index})`);
+  return {
+    content: await sms.$('p[test-id="sms-content"]').getText(),
+    state: await sms.$('.state').getText(),
+  };
+};
+
+const searchSelect = async (recipient, option) => {
+  await (await recipientField()).setValue(recipient);
+  const recipientOption = await (await $('.select2-results__options')).$(`//*[text()='${option}']`);
+  await recipientOption.waitForClickable();
+  await (await recipientOption).click();
+};
+
+
+const sendMessage = async (message, recipient, entryText) => {
+  await commonPage.clickFastActionFlat({ waitForList: false });
+  await (await $(SEND_MESSAGE_MODAL)).waitForDisplayed();
+  await searchSelect(recipient, entryText);
+  await (await messageText()).setValue(message);
   await (await sendMessageModalSubmit()).waitForClickable();
   await (await sendMessageModalSubmit()).click();
+  await $(SEND_MESSAGE_MODAL).waitForDisplayed({ reverse: true });
 };
 
-const sendMessageToPhone = async (message, phone) => {
-  await commonPage.clickFastActionFlat({ waitForList: false });
-  await messageText(message);
-  await searchSelect(phone);
-  await submitMessage();
+const sendReplyNewRecipient = async (recipient, entryText) => {
+  await searchSelect(recipient, entryText);
+  await (await sendMessageModalSubmit()).waitForClickable();
+  await (await sendMessageModalSubmit()).click();
+  await $(SEND_MESSAGE_MODAL).waitForDisplayed({ reverse: true });
 };
 
-const searchSelect = async (searchText) => {
-  await messageRecipientSelect().setValue(searchText);
-  await (await contactNameSelector()).waitForClickable(5000);
-  await (await contactNameSelector()).click();
+const sendMessageToContact = async (message) => {
+  await (await messageText()).setValue(message);
+  await (await sendMessageModalSubmit()).waitForClickable();
+  await (await sendMessageModalSubmit()).click();
+  await $(SEND_MESSAGE_MODAL).waitForDisplayed({ reverse: true });
```

Comment:
Minor, but these three lines are repeated in the previous 2 functions.  A good candidate to extract out into a shared function!

Reply:
You are completely right. Done! 🙂